### PR TITLE
add cansavecontextkey

### LINKED ISSUE #60801: Cannot Capture File->Save menu option when using a webview
We have a use case where we have created our own extension to open certain file extensions via a webview.  This basically allows the user to enter meta data stored in json in a more visual way.  One issue we are running into though is since we are using a webview, if the user selects File->Save or any of those save commands, we aren't able to capture that.

We need an event to get fired if the user does that and a WebView is the current active editor.

### PR DESCRIPTION
**Pull request by sbatten**
@bpasero This is probably not the right condition for the key, but you mentioned adding API on the editors to have the workbench question if they are save-able. Could you expand on how/where to add that or feel free to hijack that part. In any case, I think this is the right place to add and mange the new context key.

fixes  #60801

### CODE REVIEW

**Review by bpasero — CHANGES_REQUESTED** [review]
I think this is the right direction. I suggest you literally look at the conditions of the save command:

https://github.com/microsoft/vscode/blob/44ac060/src/vs/workbench/contrib/files/browser/fileCommands.ts#L119

And make the context key respect those conditions in the same way. Typically you can save any resource that our file system can handle (including contributed file systems from extensions). This includes binary files as well, not just text editors! On top of that, this includes the `untitled://` scheme. 

Also, you should not add the condition that the editor has to be dirty because it is perfectly valid to save an editor even if it is not dirty. This will touch the file, changes its modification time and e.g. can trigger save participants that still result in a change.

**Comment by sbatten** [comment]
@bpasero so i've updated the logic for this context key and it works as expected. Unfortunately while diving into the save command and inbox tracking, I've discovered a pitfall to this change. 

You can use multi-select in the explorer to save multiple files. Currently, to do this without the keybinding, some users are using the application menu entry. The recent issue I saw is that the save command should be in the explorer context menu since the application menu feels too separate from your multi-select context. 

Taking that one step further, if we make that change, the application menu's save action could refer specifically to saving editors making this PR valid. Thoughts?

File: src/vs/workbench/browser/contextkeys.ts [hunk 1]
```diff
@@ -214,8 +219,13 @@ export class WorkbenchContextKeysHandler extends Disposable {
 
 		if (activeControl) {
 			this.activeEditorContext.set(activeControl.getId());
+
+			const resource = activeControl.input.getResource();
```
Comment (bpasero) [inline src/vs/workbench/browser/contextkeys.ts]: @sbatten this fails for diff editors. Use `toResource` (`src/vs/workbench/common/editor.ts`) with the option `supportSideBySide` to support it.

File: src/vs/workbench/common/editor.ts [hunk 2]
```diff
@@ -22,6 +22,7 @@ import { IPathData } from 'vs/platform/windows/common/windows';
 import { coalesce } from 'vs/base/common/arrays';
 
 export const ActiveEditorContext = new RawContextKey<string | null>('activeEditor', null);
+export const CanSaveActiveEditorContext = new RawContextKey<boolean>('canSaveActiveEditor', false);
```
Comment (bpasero) [inline src/vs/workbench/common/editor.ts]: @sbatten I am not 100% sure this is the best name. Checking https://code.visualstudio.com/docs/getstarted/keybindings#_when-clause-contexts it seems to me we would want the thing first ("active editor"). How about `activeEditorIsSaveable` or `activeEditorCanSave`?

**Review by bpasero — CHANGES_REQUESTED** [review]

**Review by bpasero — APPROVED** [review]

**Pull request merged**